tools/fast/bin/localnet/main.go

```diff
@@ -0,0 +1,350 @@
+		collateral := big.NewInt(500)           // FIL
+		price := big.NewFloat(0.000000001)      // price per byte/block
+		expiry := big.NewInt(24 * 60 * 60 / 30) // ~24 hours
+
+		ask, err := series.CreateMinerWithAsk(ctx, miner, pledge, collateral, price, expiry)
+		if err != nil {
+			exitcode = handleError(err, "failed series.CreateMinerWithAsk;")
+			return
+		}
+
+		var data bytes.Buffer
+		dataReader := io.LimitReader(rand.Reader, int64(sectorSize))
+		dataReader = io.TeeReader(dataReader, &data)
+		_, deal, err := series.ImportAndStore(ctx, genesis, ask, files.NewReaderFile(dataReader))
+		if err != nil {
+			exitcode = handleError(err, "failed series.ImportAndStore;")
+			return
+		}
+
+		deals = append(deals, deal)
+
+	}
+
+	for _, deal := range deals {
+		err = series.WaitForDealState(ctx, genesis, deal, storagedeal.Posted)
+		if err != nil {
+			exitcode = handleError(err, "failed series.WaitForDealState;")
+			return
+		}
+	}
+
+	if shell {
+		client, err := env.NewProcess(ctx, lpfc.PluginName, options, fastenvOpts)
+		if err != nil {
+			exitcode = handleError(err, "failed to create client process;")
+			return
+		}
+
+		err = series.InitAndStart(ctx, client)
+		if err != nil {
+			exitcode = handleError(err, "failed series.InitAndStart;")
+			return
+		}
+
+		err = series.Connect(ctx, genesis, client)
+		if err != nil {
+			exitcode = handleError(err, "failed series.Connect;")
+			return
+		}
+
+		err = series.SendFilecoinDefaults(ctx, genesis, client, fil)
+		if err != nil {
+			exitcode = handleError(err, "failed series.SendFilecoinDefaults;")
+			return
+		}
+
+		interval, err := client.StartLogCapture()
+		if err != nil {
+			exitcode = handleError(err, "failed to start log capture;")
+			return
+		}
+
+		if err := client.Shell(); err != nil {
+			exitcode = handleError(err, "failed to run client shell;")
+			return
+		}
+
+		interval.Stop()
+		fmt.Println("===================================")
+		fmt.Println("===================================")
+		io.Copy(os.Stdout, interval) // nolint: errcheck
+		fmt.Println("===================================")
+		fmt.Println("===================================")
+	}
+
+	fmt.Println("Finished!")
+	fmt.Println("Ctrl-C to handleError")
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+
+	<-signals
+}
+
+func handleError(err error, msg ...string) int {
+	if err == nil {
+		return 0
+	}
+
+	if len(msg) != 0 {
+		fmt.Println(msg[0], err)
+	} else {
+		fmt.Println(err)
+	}
+
+	return 1
+}
+
+// https://stackoverflow.com/a/3070891://stackoverflow.com/a/30708914
+func isEmpty(name string) (bool, error) {
+	f, err := os.Open(name)
+	if err != nil {
+		return false, err
+	}
+	defer f.Close() // nolint: errcheck
+
+	_, err = f.Readdirnames(1) // Or f.Readdir(1)
+	if err == io.EOF {
+		return true, nil
+	}
+	return false, err // Either not empty or error, suits both cases
+}
+
+func getSectorSize(smallSectors bool) (uint64, error) {
```

Comment:
zomg this is such a painful way to do this. we should have an issue to surface sector size in plumbing as part of a params() call. fee free to just go do it, or file an issue on rose to do it after she is done removing her bits of the node api...

Reply:
If this is added to the plumbing, would it still be possible to execute that plumbing in a stand alone way that has minimal requirement such as the way this function does? This needs to be executed prior to the initialization or starting of any filecoin process. We could move this implementation, but my guess is that it would still require the same process.

Another approach would be to provide a simpler way to get the value `GetMaxUserBytesPerStagedSector` without requiring the setup of `RustSectorBuilderConfig` that only requires providing the `sectorStoreType`. That method could then be exposed through plumbing as well.

@laser do you know if it would be possible to get this value through a more minimal process?